File: test/helpers/BaseServlet.java
// Copyright (c) 2018 Exam Consortium
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package helpers;

import jakarta.servlet.http.HttpServlet;
import net.jodah.concurrentunit.Waiter;

public class BaseServlet extends HttpServlet {

    protected static String calledMethod;
    protected Waiter waiter;

    public String getLastCallMethod() {
        String call = calledMethod;
        calledMethod = null;
        return call;
    }

    public void setWaiter(Waiter waiter) {
        this.waiter = waiter;
    }

    public Waiter getWaiter() {
        return waiter;
    }
}

File: test/helpers/BaseServlet.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package helpers

import jakarta.servlet.http.HttpServlet

import java.util.concurrent.Semaphore
import scala.compiletime.uninitialized

class BaseServlet extends HttpServlet:

  protected var waiter: Semaphore = uninitialized

  def getLastCallMethod: String =
    val call = BaseServlet.calledMethod
    BaseServlet.calledMethod = null
    call

  def setWaiter(waiter: Semaphore): Unit = this.waiter = waiter
  def getWaiter: Semaphore               = waiter

object BaseServlet:
  var calledMethod: String = uninitialized
